src/domain/proof-of-liabilities/create-liabilities.ts

```diff
@@ -0,0 +1,75 @@
+import { LedgerService } from "@services/ledger"
+import { toSats } from "@domain/bitcoin"
+
+const randomInt = (min: number, max: number): number => {
+  return Math.floor(Math.random() * (max - min + 1)) + min
+}
+
+export const CreateLiabilities = async (
+  wallets: Wallet[],
+): Promise<Liabilities[] | LedgerServiceError> => {
+  const liabilities: Liabilities[] = []
+  for (const wallet of wallets) {
+    const balance = await LedgerService().getWalletBalance(wallet.id)
+    if (balance instanceof Error) return balance
+    const liability: Liabilities = {
+      accountId: wallet.accountId,
+      walletBalance: balance,
+      walletId: wallet.id,
+    }
+    liabilities.push(liability)
+  }
+  const stretchedLiabilities = getStretchedLiabilities(
```

Comment:
You should potentially shuffle before stretching, otherwise someone could predictably know how many leaves a wallet would have.